docs/cue/reference/components.cue

```diff
@@ -683,6 +692,52 @@ components: {
 				}
 			}
 
+			_proxy: {
+				common:      false
+				description: "Configures an HTTP(S) proxy for Vector to use."
+				required:    false
+				type: object: options: {
+					enabled: {
+						common:      false
+						description: "If false the proxy will be disabled."
+						required:    false
+						type: bool: default: true
+					}
+					http: {
+						common:      false
+						description: "The URL to proxy HTTP requests through."
+						required:    false
+						type: string: {
+							default: null
+							examples: ["http://foo.bar:3128"]
+							syntax: "literal"
+						}
+					}
+					https: {
+						common:      false
+						description: "The URL to proxy HTTPS requests through."
+						required:    false
+						type: string: {
+							default: null
+							examples: ["http://foo.bar:3128"]
+							syntax: "literal"
+						}
+					}
+					no_proxy: {
```

Comment:
I had a comment on a previous commit but it doesn't show up now so I'll repeat it. I am concerned this name doesn't reflect the purpose of this setting. I recommend something like `bypass` or `no_proxy_for`. @jszwedko thoughts?

Reply:
Yeah, that's a good question. I like `no_proxy` because it matches the commonly used name for this configuration (and the environment variable name), but it is a bit opaque for users that are unfamiliar with `no_proxy` from other contexts.

My inclination is to leave it as-is since `no_proxy` seems to be ubiquitous for proxy configuration.

Reply:
Fair enough, if there is precedent then there is no need to change it.